internal/node/test/BUILD.bazel

```diff
@@ -183,3 +181,24 @@ nodejs_test(
     ],
     entry_point = "npm_package_bin.spec.js",
 )
+
+# must be run with --platforms=@build_bazel_rules_nodejs//toolchains/node:linux_amd64
+nodejs_toolchain_test(
+    name = "nodejs_toolchain_linux_test",
+    expected_node_path = "external/nodejs_linux_amd64/bin/nodejs/bin/node",
+    tags = ["manual"],
+)
+
+# must be run with --platforms=@build_bazel_rules_nodejs//toolchains/node:darwin_amd64
+nodejs_toolchain_test(
+    name = "nodejs_toolchain_darwin_test",
+    expected_node_path = "external/nodejs_darwin_amd64/bin/nodejs/bin/node",
+    tags = ["manual"],
+)
+
+# must be run with --platforms=@build_bazel_rules_nodejs//toolchains/node:windows_amd64
+nodejs_toolchain_test(
+    name = "nodejs_toolchain_windows_test",
+    expected_node_path = "external/nodejs_windows_amd64/bin/nodejs/node.exe",
```

Comment:
the path seems a bit brittle as the assertion to make here. what about running `node --version` or something to assert things are working?

Reply:
what is being tested is that `ctx.toolchains["@build_bazel_rules_nodejs//toolchains/node:toolchain_type"].nodeinfo.tool_files[0].path` is set to the correct value for each platform choice. not possible to execute as the node binary will not necessarily match the host platform here.

It is a bit brittle with the path. There might be a way to derive the expected path from the platform itself if it is available in the ctx.

Reply:
Cleaned up the API for the test to make it less brittle:
```
[nodejs_toolchain_test(
    name = "nodejs_toolchain_%s_test" % platform,
    platform = platform,
    # must be run with --platforms=@build_bazel_rules_nodejs//toolchains/node:PLATFORM
    # where PLATFORM is one of BUILT_IN_NODE_PLATFORMS
    tags = ["manual"],
) for platform in BUILT_IN_NODE_PLATFORMS]
```